### PR TITLE
Expanding the config domain model

### PR DESCRIPTION
**Pull request by jadelkhoury**
## Description

Expanding the config domain model

## Checklist

- [x] This PR adds an entry to the CHANGELOG. _See https://keepachangelog.com/en/1.0.0/ for instructions. Minor edits are exempt._
- [ ] This PR was tested on at least one Lyo OSLC server or adds unit/integration tests.

### CODE REVIEW

**Comment by jadelkhoury** [comment]
For the SysML implementation, I needed parts of the config model. I have not yet tested the model with a server yet. but the generated code builds. 
I am happy to wait until I have implemented SysML to merge, but thought it good you can review this already.
The graphical model looks like this. It almost maatches the UML model in the primer. https://oslc-op.github.io/oslc-specs/notes/config-primer/config-primer.html#uml-class-diagram Where it differs, I believe the primer is inaccurate.


Configuration Management

**Review by jamsden — APPROVED** [review]
The model looks good to me.

**Pull request merged**
